cannon/mipsevm/exec/mips_instructions.go

```diff
@@ -569,3 +569,47 @@ func HandleRd(cpu *mipsevm.CpuScalars, registers *[32]Word, storeReg Word, val W
 	cpu.NextPC = cpu.NextPC + 4
 	return nil
 }
+
+func LoadSubWord(memory *memory.Memory, addr Word, byteLength Word, signExtend bool, memoryTracker MemTracker) Word {
```

Comment:
These utils are probably only useful if we want to reuse them for other load/store operations.  On the plus side, I think it makes the code easier to reason about (esoteric bit manipulation is centralized and well-tested).  But it does make the code more complex in other ways.  Open to discussing.